src/main/sensors/rangefinder.c

```diff
@@ -207,6 +270,28 @@ void rangefinderProcess(float cosTiltAngle)
             // Invalid response / hardware failure
             rangefinder.rawAltitude = RANGEFINDER_HARDWARE_FAILURE;
         }
+
+        rangefinder.snr = computePseudoSnr(distance);
+
+        if (rangefinder.snrThresholdReached == false && rangefinder.rawAltitude > 0) {
```

Comment:
Currently this calculates "noise" altitude only once per battery. I don't think its the best approach.